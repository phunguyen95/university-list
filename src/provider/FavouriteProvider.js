import React, { useState, useEffect } from "react";
import _isEmpty from "lodash/isEmpty";
export const favouriteUniList = React.createContext();

const FavouriteProvider = (props) => {
  const [favouriteList, setFavouriteList] = useState([]);
  return (
    <favouriteUniList.Provider value={{ favouriteList, setFavouriteList }}>
      {props.children}
    </favouriteUniList.Provider>
  );
};
export default FavouriteProvider;
