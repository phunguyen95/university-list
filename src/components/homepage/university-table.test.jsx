import { render, screen } from "@testing-library/react";
import UniversityTable from "./university-table";
import FavouriteProvider from "../../provider/FavouriteProvider";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

test("renders university table without crashing", () => {
  const history = createMemoryHistory();
  const favouriteList = [];
  const setFavouriteList = jest.fn();
  const handlePagination = jest.fn();
  const currentPageIndex = 0;
  const handlePrevPage = jest.fn();
  const handleNextPage = jest.fn();
  const handleOnSelect = jest.fn();
  const { asFragment } = render(
    <Router history={history}>
      <FavouriteProvider value={{ favouriteList, setFavouriteList }}>
        <UniversityTable
          data={[]}
          isSearching={false}
          totalCount={0}
          handlePagination={handlePagination}
          currentPageIndex={currentPageIndex}
          handlePrevPage={handlePrevPage}
          handleNextPage={handleNextPage}
          handleOnSelect={handleOnSelect}
        />
      </FavouriteProvider>
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});
test("renders university table with uniInfo", () => {
  const history = createMemoryHistory();
  const favouriteList = [
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
  ];
  const setFavouriteList = jest.fn();
  const handlePagination = jest.fn();
  const currentPageIndex = 1;
  const handlePrevPage = jest.fn();
  const handleNextPage = jest.fn();
  const handleOnSelect = jest.fn();
  const { asFragment } = render(
    <Router history={history}>
      <FavouriteProvider value={{ favouriteList, setFavouriteList }}>
        <UniversityTable
          data={favouriteList}
          isSearching={false}
          totalCount={1}
          handlePagination={handlePagination}
          currentPageIndex={currentPageIndex}
          handlePrevPage={handlePrevPage}
          handleNextPage={handleNextPage}
          handleOnSelect={handleOnSelect}
        />
      </FavouriteProvider>
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});
test("renderPaginationOption with more than 11 elements", () => {
  const history = createMemoryHistory();
  const favouriteList = [
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
    {
      name: "Uni 1",
      country: "VN",
      web_pages: ["abc.xyz"],
    },
  ];
  const setFavouriteList = jest.fn();
  const handlePagination = jest.fn();
  const currentPageIndex = 1;
  const handlePrevPage = jest.fn();
  const handleNextPage = jest.fn();
  const handleOnSelect = jest.fn();
  const { asFragment } = render(
    <Router history={history}>
      <FavouriteProvider value={{ favouriteList, setFavouriteList }}>
        <UniversityTable
          data={favouriteList}
          isSearching={false}
          totalCount={240}
          handlePagination={handlePagination}
          currentPageIndex={currentPageIndex}
          handlePrevPage={handlePrevPage}
          handleNextPage={handleNextPage}
          handleOnSelect={handleOnSelect}
        />
      </FavouriteProvider>
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
