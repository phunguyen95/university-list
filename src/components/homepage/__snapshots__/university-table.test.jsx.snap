// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderPaginationOption with more than 11 elements 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <h4>
      Total 240 seach results
    </h4>
    <label
      class="label sortBy"
    >
      Sort By:
    </label>
    <div
      class="select is-normal"
    >
      <select>
        <option
          value="country"
        >
          Country
        </option>
        <option
          selected=""
          value="name"
        >
          Name
        </option>
      </select>
    </div>
    <div
      class="level-item"
    >
      <small>
        Page 1 of 12 
      </small>
    </div>
    <div
      class="container"
    >
      <div
        class="b-table has-pagination"
      >
        <div
          class="table-wrapper has-mobile-cards"
        >
          <table
            class="table is-fullwidth is-striped is-hoverable is-fullwidth"
          >
            <thead>
              <tr>
                <th>
                  No.
                </th>
                <th>
                  Name
                </th>
                <th>
                  Country
                </th>
                <th>
                  Website
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  data-label="No"
                >
                  1
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  2
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  3
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  4
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  5
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  6
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  7
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  8
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  9
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  10
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  11
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
              <tr>
                <td
                  data-label="No"
                >
                  12
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="notification"
        >
          <div
            class="level"
          >
            <div
              class="level-left"
            >
              <div
                class="level-item"
              >
                <div
                  class="buttons has-addons"
                >
                  <button
                    class="button is-current"
                    type="button"
                  >
                    1
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    2
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    3
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    4
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    5
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    6
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    7
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    8
                  </button>
                  <button
                    class="button"
                    type="button"
                  >
                    9
                  </button>
                  <span
                    class="pagination-ellipsis"
                  >
                    â€¦
                  </span>
                  <button
                    class="button"
                    type="button"
                  >
                    12
                  </button>
                </div>
              </div>
              <div
                class="level-right"
              >
                <button
                  class="pagination-previous"
                >
                  Previous
                </button>
                <button
                  class="pagination-next"
                >
                  Next page
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders university table with uniInfo 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <h4>
      Total 1 seach results
    </h4>
    <label
      class="label sortBy"
    >
      Sort By:
    </label>
    <div
      class="select is-normal"
    >
      <select>
        <option
          value="country"
        >
          Country
        </option>
        <option
          selected=""
          value="name"
        >
          Name
        </option>
      </select>
    </div>
    <div
      class="level-item"
    >
      <small>
        Page 1 of 1 
      </small>
    </div>
    <div
      class="container"
    >
      <div
        class="b-table has-pagination"
      >
        <div
          class="table-wrapper has-mobile-cards"
        >
          <table
            class="table is-fullwidth is-striped is-hoverable is-fullwidth"
          >
            <thead>
              <tr>
                <th>
                  No.
                </th>
                <th>
                  Name
                </th>
                <th>
                  Country
                </th>
                <th>
                  Website
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  data-label="No"
                >
                  1
                </td>
                <td
                  data-label="Name"
                >
                  Uni 1
                </td>
                <td
                  data-label="Country"
                >
                  VN
                </td>
                <td
                  data-label="Website"
                >
                  abc.xyz
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="notification"
        >
          <div
            class="level"
          >
            <div
              class="level-left"
            >
              <div
                class="level-item"
              >
                <div
                  class="buttons has-addons"
                />
              </div>
              <div
                class="level-right"
              >
                <button
                  class="pagination-previous"
                  disabled=""
                >
                  Previous
                </button>
                <button
                  class="pagination-next"
                >
                  Next page
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders university table without crashing 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <h4>
      Total 0 seach results
    </h4>
    <label
      class="label sortBy"
    >
      Sort By:
    </label>
    <div
      class="select is-normal"
    >
      <select>
        <option
          value="country"
        >
          Country
        </option>
        <option
          selected=""
          value="name"
        >
          Name
        </option>
      </select>
    </div>
    <div
      class="level-item"
    >
      <small>
        Page 0 of 1 
      </small>
    </div>
    <div
      class="container"
    >
      <div
        class="b-table"
      >
        <div
          class="table-wrapper has-mobile-cards"
        >
          <table
            class="table is-fullwidth is-striped is-hoverable is-fullwidth"
          >
            <thead>
              <tr>
                <th>
                  No.
                </th>
                <th>
                  Name
                </th>
                <th>
                  Country
                </th>
                <th>
                  Website
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="is-empty"
              >
                <td
                  colspan="7"
                >
                  <section
                    class="section"
                  >
                    <div
                      class="content has-text-grey has-text-centered"
                    >
                      <p>
                        <span
                          class="icon is-large"
                        >
                          <i
                            class="mdi mdi-emoticon-sad mdi-48px"
                          />
                        </span>
                      </p>
                      <p>
                        Nothing's thereâ€¦
                      </p>
                    </div>
                  </section>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="notification"
        >
          <div
            class="level"
          >
            <div
              class="level-left"
            >
              <div
                class="level-item"
              >
                <div
                  class="buttons has-addons"
                />
              </div>
              <div
                class="level-right"
              >
                <button
                  class="pagination-previous"
                  disabled=""
                >
                  Previous
                </button>
                <button
                  class="pagination-next"
                  disabled=""
                >
                  Next page
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
